### PR TITLE
track HEC IDs per request ID

### PR DESCRIPTION
**Pull request by john-floren-gravwell**
Addresses #608

### CODE REVIEW

File: ingesters/HttpIngester/hec.go [hunk 1]
```diff
@@ -169,13 +170,23 @@ loop:
...
 }
 
+func (hh *hecHandler) writeAck(channel string, w http.ResponseWriter) {
+	hh.ackLock.Lock()
+	defer hh.ackLock.Unlock()
+	if _, ok := hh.ackIds[channel]; !ok {
+		hh.ackIds[channel] = 0
+	}
+	hh.ackIds[channel] = hh.ackIds[channel] + 1
+	json.NewEncoder(w).Encode(ack{
```
Comment (kris-watts-gravwell) [inline ingesters/HttpIngester/hec.go:185]: I think we need to release the acklock before writing to the network.  A slow client would block all other clients

**Review by kris-watts-gravwell — CHANGES_REQUESTED** [review]
Need to release lock before going to the network on the writeAck command

**Comment by john-floren-gravwell** [comment]
ptal

**Review by kris-watts-gravwell — APPROVED** [review]
LGTM

**Pull request merged**